OneCommunityGlobal/HighestGoodNetworkApp
# Olena Lesson Card Functionality

**Description**
Lesson list component for Phase II
<img width="680" alt="lessonlistpage" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/49422649/28583fa9-f2fe-45f0-b12d-1051b1137a99">


**Related PRs (if any):**
This frontend PR is related to thehttps://github.com/OneCommunityGlobal/HGNRest/pull/716 backend PR (ONLY to test LIKE functionality)
&&
https://github.com/OneCommunityGlobal/HGNRest/pull/715 backend PR (Other functionality like EDIT, DELITE Card, FILTER, SORT)
…

**Main changes explained:**
Implemented functionality for the Edit & Delete buttons on the Lesson Card (visible and operable only for the author of the card).
Users can click on the Edit button to modify the lesson summary, with added validation to ensure the summary is not empty.
Users can click on the Delete button to remove the entire card.



**How to test:**

1. Checkout to the current branch.
2. Run npm install and ... to run this PR locally.
3. Log in as any user.
4. Go to http://localhost:3000/bmdashboard/lessonlist.
5. localStorage.clear() in the console
6. Verify the design, tag input functionality, and expand/collapse card functionality.
7. Test the Edit & Delete buttons, Like button functionality and Sort/Tag filter functionality and overall page design. To test, please change the authorId to your userID in backend file db.js.

**## Note:**
Please note that the Figma navigation bar and the current design differ. I've made adjustments based on Jae's request, and he has also verified the design.


## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/49422649/e2d6ca0e-8420-4a48-aaa8-1aba58030c0e




https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/49422649/f27754cd-e4bf-4739-a348-a59c13861e6b
